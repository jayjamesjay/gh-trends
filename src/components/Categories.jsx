import React from 'react';
import PropTypes from 'prop-types';

import { CategoryMenu } from '../styles/Form';
import Category from './Category';

/**
 * @module Categories
 */

/**
 * Menu to hadle category selection
 *
 * @property {array} labels - names of categories
 * @property {string} active - name of label which is currently active
 * @property {function} onClick - function called whenever label is clicked
 * @returns {Categories}
 */
export default function Categories({ labels, active, onClick }) {
  return (
    <CategoryMenu>
      {labels.map((label) => (
        <Category checked={active === label} key={label} label={label} onClick={onClick} />
      ))}
    </CategoryMenu>
  );
}

Categories.propTypes = {
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
  active: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
