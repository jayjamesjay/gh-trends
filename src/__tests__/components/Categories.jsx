import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Categories from '../../components/Categories';

describe('<Categories />', () => {
  const labels = ['Year', 'Month'];
  const active = 'Month';
  const func = () => {};

  it(`renders with default style`, () => {
    const component = renderer.create(
      <Categories labels={labels} active={active} onClick={func} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`displays labels`, () => {
    render(<Categories labels={labels} active={active} onClick={func} />);

    expect(screen.getByText(labels[0]).textContent).toEqual(labels[0]);
    expect(screen.getByText(labels[1]).textContent).toEqual(labels[1]);
  });

  it(`checks active category`, () => {
    render(<Categories labels={labels} active={active} onClick={func} />);
    expect(screen.getAllByRole('radio', { checked: true })[1].value).toEqual(labels[1]);
  });
});
