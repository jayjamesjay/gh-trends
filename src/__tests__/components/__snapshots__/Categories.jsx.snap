// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders with default style 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  text-transform: uppercase;
}

.c2 {
  position: relative;
  z-index: 100;
}

.c3 {
  opacity: 0;
}

.c1 {
  display: inherit;
  padding: 1rem 1.5rem;
  position: relative;
  background: transparent;
  overflow: hidden;
  font-size: 1.3rem;
  cursor: pointer;
}

.c1::before {
  content: '';
  width: 90%;
  height: 90%;
  position: absolute;
  top: 5%;
  left: 5%;
  z-index: 20;
  background: #fff;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  -webkit-transition: -webkit-transform 0.15s ease-in-out;
  -webkit-transition: transform 0.15s ease-in-out;
  transition: transform 0.15s ease-in-out;
  -webkit-transform: translateX(-200%);
  -ms-transform: translateX(-200%);
  transform: translateX(-200%);
}

.c4 {
  display: inherit;
  padding: 1rem 1.5rem;
  position: relative;
  background: transparent;
  overflow: hidden;
  font-size: 1.3rem;
  cursor: pointer;
}

.c4::before {
  content: '';
  width: 90%;
  height: 90%;
  position: absolute;
  top: 5%;
  left: 5%;
  z-index: 20;
  background: #fff;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  -webkit-transition: -webkit-transform 0.15s ease-in-out;
  -webkit-transition: transform 0.15s ease-in-out;
  transition: transform 0.15s ease-in-out;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

<form
  className="c0"
>
  <div
    aria-checked={false}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="radio"
    tabIndex="0"
  >
    <label
      className="c1"
      htmlFor="Year"
    >
      <span
        className="c2"
      >
        Year
      </span>
    </label>
    <input
      checked={false}
      className="c3"
      id="Year"
      onChange={[Function]}
      tabIndex="-1"
      type="radio"
      value="Year"
    />
  </div>
  <div
    aria-checked={true}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="radio"
    tabIndex="0"
  >
    <label
      className="c4"
      htmlFor="Month"
    >
      <span
        className="c2"
      >
        Month
      </span>
    </label>
    <input
      checked={true}
      className="c3"
      id="Month"
      onChange={[Function]}
      tabIndex="-1"
      type="radio"
      value="Month"
    />
  </div>
</form>
`;
